.azure-pipelines-steps.yml

```diff
@@ -0,0 +1,27 @@
+steps:
+- script: npm i -g npm@$(npm_version)
+  displayName: Use legacy npm version $(npm_version)
+  condition: ne(variables['npm_version'], '')
+
+- task: NodeTool@0
+  inputs:
+    versionSpec: '$(node_version)'
+  displayName: Use Node $(node_version)
+
+- script: npm install
+  displayName: npm install
+
+- script: npm run coveralls
+  env:
+    COVERALLS_REPO_TOKEN: $(COVERALLS_REPO_TOKEN_SECRET)
+  displayName: Run coveralls
+  condition: eq(variables['run_coveralls'], true)
+
+- script: npm test
+  displayName: Run tests
+  condition: ne(variables['run_coveralls'], true)
+
+- task: PublishTestResults@2
+  inputs:
+    testResultsFiles: '**/xunit.xml'
```

Comment:
So azure pipelines only speaks xunit?

Reply:
With this input it does, but it can also use JUnit, NUnit, VSTest, xUnit, or cTest, we'd just need to change testResultsFiles to something like `**/junit.xml`. I believe if we want we could also do something like `**/TEST-**.xml`, this input really just works as a pattern matcher I think.

I used xunit since that matches the custom reporter I added (more on that below).

Reply:
@damccorm It looks like mocha (even the version we use) [ships with an XUnit reporter](https://mochajs.org/#xunit).  Can we just use that?  We can have an `azure` script that does `mocha --async-only --reporter xunit`

That way we can drop the extra reporter.

Reply:
That works, the only downside is we don't get great output in the console.

What we get with multi-reporter:

![image](https://user-images.githubusercontent.com/42773683/54284731-da27a700-4576-11e9-8896-7bbf5b1a4ce4.png)

What we get with xunit reporter:

![image](https://user-images.githubusercontent.com/42773683/54284668-bfedc900-4576-11e9-88d6-17c070bcecb9.png)

I have this change in PR, if you're comfortable with that tradeoff I'll merge just let me know.

Reply:
Oh actually, on closer inspection, this doesn't get written to a file which is necessary for us to report results. We can pipe it out to a file, but that would mean we would get no output at all on the console.

Reply:
We can use the `-O output=test.xunit` flag to specify the output file.  I was thinking we could just run the test suites as separate phases, one with the spec reporter and one with the xunit reporter.

Reply:
Makes sense to me! Updated accordingly